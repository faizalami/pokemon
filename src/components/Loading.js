/** @jsxImportSource @emotion/react */
import { Flex } from './FlexGrid';
import { padding, width } from './utilities';
import Image from './Image';
import { css, keyframes } from '@emotion/react';

const rotateAnimation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`;

const loadingIconStyle = css`
  animation: ${rotateAnimation} 3s infinite linear;
`;

function Loading () {
  return (
    <Flex justifyContent="center" css={[width.full, padding.a2]}>
      <Image
        png="/assets/img/png/pokeball-64.png"
        webp="/assets/img/webp/pokeball-64.webp"
        alt="Loading"
        width={64}
        height={64}
        css={loadingIconStyle}
      />
    </Flex>
  );
}

export default Loading;
