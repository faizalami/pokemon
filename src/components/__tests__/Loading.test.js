import { render } from '@testing-library/react';
import Loading from '../Loading';

test('Loading matches snapshot', () => {
  expect(render(<Loading/>)).toMatchSnapshot();
});
