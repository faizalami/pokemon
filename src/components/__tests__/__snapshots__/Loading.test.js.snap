// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": @keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(359deg);
    -moz-transform: rotate(359deg);
    -ms-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  padding: 0.5rem;
}

.emotion-1 {
  object-fit: contain;
  object-position: center;
  -webkit-animation: animation-0 3s infinite linear;
  animation: animation-0 3s infinite linear;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <picture>
          <source
            data-testid="image-webp"
            srcset="/assets/img/webp/pokeball-64.webp"
            type="image/webp"
          />
          <source
            data-testid="image-png"
            srcset="/assets/img/png/pokeball-64.png"
            type="image/png"
          />
          <img
            alt="Loading"
            class="emotion-1"
            data-testid="image-regular"
            height="64"
            loading="lazy"
            src="/assets/img/png/pokeball-64.png"
            width="64"
          />
        </picture>
      </div>
    </div>
  </body>,
  "container": @keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(359deg);
    -moz-transform: rotate(359deg);
    -ms-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  padding: 0.5rem;
}

.emotion-1 {
  object-fit: contain;
  object-position: center;
  -webkit-animation: animation-0 3s infinite linear;
  animation: animation-0 3s infinite linear;
}

<div>
    <div
      class="emotion-0"
    >
      <picture>
        <source
          data-testid="image-webp"
          srcset="/assets/img/webp/pokeball-64.webp"
          type="image/webp"
        />
        <source
          data-testid="image-png"
          srcset="/assets/img/png/pokeball-64.png"
          type="image/png"
        />
        <img
          alt="Loading"
          class="emotion-1"
          data-testid="image-regular"
          height="64"
          loading="lazy"
          src="/assets/img/png/pokeball-64.png"
          width="64"
        />
      </picture>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
